device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/amqps/AmqpsIotHubConnection.java

```diff
@@ -1132,8 +1143,6 @@ public Object call()
             }
             catch (HandlerException e)
             {
-                iotHubReactor.free();
```

Comment:
It turns out that if an IOException is thrown due to no network, this catch block isn't run because the AMQP layer delegates the exception to the main thread instead. To help cover similar cases where the reactor thread encounters but does not throw an exception, it makes more sense for this free() to be in the finally block.